staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -224,7 +230,10 @@ func (flags *ApplyFlags) AddFlags(cmd *cobra.Command) {
 	cmd.Flags().BoolVar(&flags.Overwrite, "overwrite", flags.Overwrite, "Automatically resolve conflicts between the modified and live configuration by using values from the modified configuration")
 	cmd.Flags().BoolVar(&flags.Prune, "prune", flags.Prune, "Automatically delete resource objects, that do not appear in the configs and are created by either apply or create --save-config. Should be used with either -l or --all.")
 	cmd.Flags().BoolVar(&flags.All, "all", flags.All, "Select all resources in the namespace of the specified resource types.")
-	cmd.Flags().StringArrayVar(&flags.PruneWhitelist, "prune-whitelist", flags.PruneWhitelist, "Overwrite the default whitelist with <group/version/kind> for --prune")
+	cmd.Flags().StringArrayVar(&flags.PruneAllowlist, "prune-allowlist", flags.PruneAllowlist, "Overwrite the default allowlist with <group/version/kind> for --prune")
+	cmd.Flags().StringArrayVar(&flags.PruneWhitelist, "prune-whitelist", flags.PruneWhitelist, "Overwrite the default whitelist with <group/version/kind> for --prune") // TODO: Remove this in kubectl 1.29 or later
+	cmd.Flag("prune-whitelist").Deprecated = "Use --prune-allowlist instead."
+	cmd.Flag("prune-whitelist").Hidden = true
```

Comment:
I see there is a `cmd.Flags().MarkDeprecated` that does both deprecated and hidden at the same time. We use that in three places, which suggests doing both at once is indeed standard (makes sense to me as well--the flag will continue to work for folks already using it, but folks consulting the help won't newly discover it). Any reason not to use that method here?

Reply:
Ah thanks for that.  I wasn't aware of `MarkDeprecated`.  I'll change the code to use it.